cpp/include/cugraph/detail/shuffle_wrappers.hpp

```diff
@@ -62,6 +62,25 @@ template <typename vertex_t>
 rmm::device_uvector<vertex_t> shuffle_ext_vertices_by_gpu_id(
   raft::handle_t const& handle, rmm::device_uvector<vertex_t>&& d_vertices);
 
+/**
+ * @brief Shuffle vertex/value tuples using the external vertex key function which returns the
+ * target GPU ID.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam value_t Type of values.
+ *
+ * @param[in] handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator,
+ * @param[in] d_vertices Vertex IDs to shuffle
+ * @param[in] d_values Values to shuffle
+ *
+ * @return tuple containing device vector of shuffled vertices and device vector of shuffled values
+ */
+template <typename vertex_t, typename value_t>
+std::tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<value_t>>
+shuffle_ext_vertices_and_values_by_gpu_id(raft::handle_t const& handle,
```

Comment:
So, we have collect_local_vertex_values_from_ext_vertex_value_pairs (https://github.com/rapidsai/cugraph/pull/2467/files#diff-08a102118cad7e0349c4888019ea4b596112841ec316c0ed93f0542ac55f01c3R156).

Should we better rename this to `shuffle_ext_vertex_value_pairs_by_gpu_id` to be more consistent in naming?

Reply:
The behavior is subtly different, the shuffle functions take a set of inputs and return that set of inputs shuffled to the proper GPU.  The collect function does a shuffle operation, but then executes other logic (in this case a `thrust::scatter`) to generate the desired output.  So I'm not sure that using the same naming pattern is semantically meaningful.  Perhaps `shuffle_scatter_ext_vertex_value_pairs_by_gpu_id` or something like that, although I'm not sure that's particularly meaningful either.

Feels like a thing to defer to later, but perhaps to keep in mind as we review those functions.

Reply:
I will take a closer look, but this function internally calls `cugraph::groupby_gpu_id_and_shuffle_kv_pairs`. So, I just assumed that this shuffle key, value pairs and where `key == external vertex ID`. `external_vertex_value_pairs` may sound more consistent with `kv_pairs`.

Anyways, not something super critical, so I'm approving if this needs to be merged faster for this release.